test/ruby/test_gc_compact.rb

```diff
@@ -270,6 +270,9 @@ def add_ivars
       ary = OBJ_COUNT.times.map { Foo.new }
       ary.each(&:add_ivars)
 
+      GC.start
```

Comment:
This test is failing with our change, because without making sure that the necessary shapes are already allocated under the new size pools root shape, our code will decide not to move the objects so as to avoid allocating during GC.

We added this code to force the creation of the shape tree, to allow the objects to move during this test